cluster-autoscaler/cloudprovider/clusterapi/clusterapi_nodegroup.go

```diff
@@ -265,3 +335,11 @@ func newNodegroupFromMachineDeployment(controller *machineController, machineDep
 		scalableResource:  scalableResource,
 	}, nil
 }
+
+func buildGenericLabels(nodeName string) map[string]string {
```

Comment:
Might be worth a comment to explain what these labels are and why we are adding them? Like, why did we choose these three in particular to add?

Reply:
so, i looked around a little bit and i see these values used by some of the other cloud providers for the autoscaler, but i don't see them used in all of the cluster-api provider implementations.

at a speculation, @frobware might have copied these as they are present in the other providers but they are not directly used by the cluster-api implementation.

i will add a TODO here, i think we should probably re-assess their inclusion after further testing.